cmd/difference.go

```diff
@@ -232,8 +243,14 @@ func differenceInternal(sourceURL string, srcCh <-chan *ClientContent, targetURL
 		tgtEOF = !tgtOk
 
 		// No objects from source AND target: Finish
-		if srcEOF && tgtEOF {
-			break
+		if opts.isRemove {
+			if srcEOF && tgtEOF {
+				break
+			}
+		} else {
+			if srcEOF {
+				break
```

Comment:
in fact, we cannot exit here, 

if the source listing is finished and there are still more entries in the target listing, we will generate a wrong difference here because we are exiting early.

isn't an early quitting here when we detect a canceled context is  enough to fix the issue ?

Reply:
ok.. I think I see what you mean. I'm probably exiting to early because we need to do all the diff check stuff. 
I'll check if there is a better way.

I don't think cancel context is enough, but I'll re-visit this tomorrow

Reply:
NOTE: we ended up with a similar result, but taking into account the active-active flag.